### ISSUE TITLE
Only getting a purple screen

### ISSUE THREAD

**Issue by maeries** [pawelmalak/flame#24]
After deploying flame using this docker-compose file

```
version: '3'
services:
    flame:
        image: pawelmalak/flame:latest
        restart: unless-stopped
        volumes:
            - ./data:/app/data
        networks:
            - caddy_net

networks:
  caddy_net:
    external: true
```
I only get a purple screen when visiting the URL (I use Caddy2 as a reverse proxy). When I refresh the page I can see

```
Thursday 10, June 2021
Good Evening

Applications

Bookmarks
```
for a fraction of a second, before only the background color is shown again.

I tried Firefox and vivaldi and both show the same behaviour

**Comment by pawelmalak** [comment]
Are you getting any errors in the console?

**Comment by roo82** [comment]
I'm seeing the same thing when trying to access flame from behind an nginx reverse proxy. It works fine if I hit the site at the docker host:port itself, so I am guessing I'm missing some config or something. I can post my nginx config if that would help.

**Comment by pawelmalak** [comment]
I have never used reverse proxy before so I will try to setup one, check if I can reproduce this problem and then work from there. I suspect it all comes from this ws:// protocol.

**Comment by maeries** [comment]
Indeed, it works when not using a reverse proxy

**Comment by Lanjelin** [comment]
Nginx behind reverse proxy with https. Console spits out the following.
```Mixed Content: The page at 'https://example.com/' was loaded over HTTPS, but attempted to connect to the insecure WebSocket endpoint 'ws://example.com/socket'. This request has been blocked; this endpoint must be available over WSS.```
Suppose wss is needed to serve the content securely / via https.

Nginx using http is working perfectly with the following configuration (some of the settings could probably be skipped):
```
server {
    listen           80;
    listen      [::]:80;
    server_name home.example.com;
    location / {
        proxy_set_header X-Forwarded-Server $host;
        proxy_set_header X-Forwarded-Host $host;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_pass http://192.168.1.148:5005;
        proxy_http_version 1.1;
        proxy_pass_request_headers on;
        proxy_set_header Connection "upgrade";
        proxy_store off;
    }
}
```

**Comment by pawelmalak** [comment]
So I setup reverse proxy using NginxProxyManager and got the same error on all major browsers.
As expected, the error was caused by use of `ws://` protocol with `https://`.

I've added simple check on WeatherWidget which will connect to either `ws` or `wss` based on `document.location.protocol` so it can be used both in development and on production.

I don't know how it works with other reverse proxy services but here I had to allow for websocket support to get it fully working.
